devito/ops/nodes.py

```diff
@@ -0,0 +1,22 @@
+import numbers
+from cached_property import cached_property
+
+from devito.ir.iet import nodes
+from devito.types.basic import AbstractFunction
+
+from devito.ops.types import String
+
+
+class Call(nodes.Call):
+
+    @cached_property
+    def free_symbols(self):
```

Comment:
When we send a String as an argument, the function free_symbol does not know how to handle it.
If this need arose only for the OPS backend, we can not change the Devito's code, right?

Reply:
overriding this and adding String seems an overkill to me

I do understand the issue you encountered, though

how did @vincepandolfo do this?

I think you can subclass sympy Symbol directly for this sort of minor things. 

```
class Literal(Symbol):
    pass
```

to be added somewhere [here](https://github.com/opesci/devito/blob/master/devito/symbolics/extended_sympy.py#L358)

and then extend [here](https://github.com/opesci/devito/blob/master/devito/ir/iet/utils.py#L113) such that it does ``... isinstance(Macro, Literal)``

I'd be happy to accept these few additional lines in the codebase. Not only do they make sense, but also would allow avoiding this override as well as the String subclass

What do you think?

Reply:
My solution was identical ( https://github.com/vincepandolfo/devito/blob/master/devito/ir/iet/nodes.py#L267 )

Reply:
We are using Vincenzo's code as a reference, but we are removing the changes in the codebase.

Reply:
@FabioLuporini  Making [p for p in parameters if not isinstance(p, (Literal, Macro)), we will have:
ops_partition() instead ops_partition('').
Because the Literal was dropped.